specs/enable_gluster_native_events.adoc

```diff
@@ -0,0 +1,147 @@
+= Enable gluster native events in tendrl
+
+Events generated by glusterfs events API should be exposed by tendrl.
+
+
+== Problem description
+
+Currently tendrl is not utilizing the events exposed by glusterfs events API.
+Tendrl should be enhanced to consume and handle the events.
+
+== Use Cases
+
+* Glusterfs events API pushes events for certain internal glusterfs failure conditions
+which is not track able by tendrl. Tendrl can raise suitable alerts for such events.
+
+* The interval of sds state sync can be increased (to 30 min) and we should have event
+driven state sync. which will considerably increase the efficiency of gluster integration
+service.
+
+== Proposed change
+
+The event receiver web-hook can be run on all the gluster storage nodes as a
+thread/greenlet in gluster-integration daemon. And we should configure the
+gluster eventing daemon to push events in localhost mode (the events generated
+by gluster eventing daemon will be pushed to the web-hook running only on
+that particular node). With this we are eliminating single point of failure
+w.r.t event listener.
+
+Since some of the events are raised from all the nodes or raised multiple times
+(till alert condition gets fixed), tendrl should make sure that duplicate events
+are handled appropriately by checking the state information stored in etcd.
+
+In future, gluster integration state sync should be based on events received
+by glusterfs events API. Thus we make sure that we do the state sync only when
+there is a state change. Also we can have a 30 min sync, to make sure that we
```

Comment:
So at the moment the aggregations at cluster level like cluster  utilization, cluster health status, cluster client connections etc are done at last post volume sync. There are plans to move this aggregation logic as a flow which would be triggered while starting the `{gluster/ceph}_integration` and after first successful run, this job would self schedule at regular interval. This would be picked by one node of the cluster randomly and processed.
Hope this makes clear.